dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ThreadCreation.java

```diff
@@ -203,9 +207,13 @@ private void resolveDynamicThreadJoin(Program program, List<ThreadData> threadDa
                             EventFactory.newGoto(joinEnd)
                     );
                 } else {
+                    final Expression detachedOffset = expressions.makeOne((IntegerType) data.comAddress.getType());
                     // Successful join
                     caseBody = eventSequence(
                             joinCase,
+                            newLoad(detachedRegister, expressions.makeAdd(data.comAddress, detachedOffset)),
```

Comment:
I do not understand how `comAddress+1` may not clash with other addresses.

EDIT: after reading the whole code, I got it